### PR TITLE
forgetpassword and reset password

### PR DESCRIPTION
**Pull request by SaFaaMOSTAFAA**

### CODE REVIEW

File: users/models.py [hunk 1]
```diff
@@ -47,5 +48,11 @@ class Trader(UserAccount):
 
 
 class CustomerAccount(UserAccount):
-    email = models.EmailField()
+    # email = models.EmailField()
```
Comment (HossamAmeen) [inline users/models.py]: why you comment email ?

Comment (HossamAmeen) [reply users/models.py]: please remove it if you don't use it

File: users/serializers.py [hunk 2]
```diff
@@ -32,12 +32,33 @@ class CustomerAccountSerializer(serializers.ModelSerializer):
...
+        regex=r'^(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$',
+        write_only=True,
+        error_messages={
+            'invalid': (
+                'Password must be at least 8 characters long with '
+                'at least one capital letter and symbol'
+            )
+        }
+    )
+    confirm_password = serializers.CharField(write_only=True, required=True)
```
Comment (Hossam-Ameen) [inline users/serializers.py:60]: What about the user who submits the password that is not the same as the confirmed password?

File: users/api.py [hunk 3]
```diff
@@ -64,3 +71,59 @@ def perform_update(self, serializer):
...
+
+    def post(self, request):
+
+        email = request.data['email']
+        user = UserAccount.objects.filter(email__iexact=email).first()
+
+        if user:
+            token_generator = PasswordResetTokenGenerator()
+            token = token_generator.make_token(user)
+            expiration_date = timezone.now() + timedelta(hours=24)
```
Comment (Hossam-Ameen) [inline users/api.py]: please update this timedelta(hours=24) to be one hour

File: users/api.py [hunk 4]
```diff
@@ -64,3 +71,59 @@ def perform_update(self, serializer):
...
+
+            reset = PasswordReset(email=email,
```
Comment (Hossam-Ameen) [inline users/api.py]: why did you use .create() ?

**Pull request merged**